zoe-cli/src/commands/main.kt

```diff
@@ -182,7 +185,7 @@ fun mainModule(context: CliContext) = module {
                 EnvVarsConfigUrlProvider,
                 LocalConfigDirUrlProvider(context.configDir)
             )
-        ).createConfig(context.env)
+        ).createConfig(context.env).let { envConfig -> customize(envConfig) }
```

Comment:
Instead of injecting the customizer here, what do you think of rather overriding the `EnvConfig` bean during tests as explained in [this section of the documentation](https://doc.insert-koin.io/#/koin-core/modules?id=overriding-definition-or-module)?

We could define a testing instance of `EnvConfig` that we build from scratch in the tests and fullfill it with values from the test containers, etc. and then use it as an override.

Reply:
Yes that's definitely a better solution! I will update it accordingly 😊